tests/test_annotation_stores.py

```diff
@@ -186,7 +201,280 @@ def _fill_store(
+    assert ann._geometry is None
+    # Check that coords are the same after creating geometry
+    _ = ann.geometry
+    assert ann._geometry is not None
+    geom_coords = ann.coords
+    assert all(np.array_equal(a, b) for a, b in zip_longest(geom_coords, coords))
+
+
+def test_annotation_multi_line_string_wkb_coords() -> None:
+    """Test generating coordinates from WKB for MultiLineString.
+
+    This should produce coordinates without generating a Shapely
+    geometry.
+    """
+    multi_line_string = MultiLineString(
+        [
+            LineString([(0, 0), (1, 1)]),
+            LineString([(2, 2), (3, 3)]),
+        ],
+    )
+    ann = Annotation(wkb=multi_line_string.wkb)
+    coords = ann.coords
+    assert len(coords) == len(multi_line_string.geoms)
+    assert isinstance(coords[0], np.ndarray)
+    assert ann._geometry is None
+    # Check that coords are the same after creating geometry
+    _ = ann.geometry
+    assert ann._geometry is not None
+    geom_coords = ann.coords
+    assert all(np.array_equal(a, b) for a, b in zip_longest(geom_coords, coords))
+
+
+def test_annotation_coords_wkb_uknown_geometry() -> None:
+    """Test generating coordinates from WKB for unknown geometry type."""
+    ann = Annotation(wkb=b"\x00\x07\x00\x00\x00")
+    with pytest.raises(ValueError, match="Unknown geometry type"):
+        _ = ann.coords
+
+
+def test_annotation_coords_geometry_uknown_geometry(
+    monkeypatch,  # noqa: ARG001
+) -> None:
+    """Test generating coordinates from WKB for unknown geometry type."""
+    ann = Annotation(geometry=GeometryCollection([Point(1, 2)]))
+    with pytest.raises(ValueError, match="Unknown geometry type"):
+        _ = ann.coords
+
+
+def test_annotation_geometry_wkb() -> None:
+    """Test generating WKB from geometry."""
+    polygon = Polygon([[0, 0], [1, 1], [2, 0]])
+    ann = Annotation(geometry=polygon)
+    assert ann.wkb == polygon.wkb
+
+
+def test_annotation_geometry_geometry_type() -> None:
+    """Test getting GeometryType from Shapely geometry."""
+    polygon = Polygon([[0, 0], [1, 1], [2, 0]])
+    ann = Annotation(geometry=polygon)
+    assert ann.geometry_type == GeometryType.POLYGON
+
+
+def test_annotation_hash() -> None:
+    """Test that the hash is the same for two annotations with the same geometry."""
+    ann_1 = Annotation(Polygon([[0, 0], [1, 1], [2, 0]]))
+    ann2 = Annotation(Polygon([[0, 0], [1, 1], [2, 0]]))
+    assert hash(ann_1) == hash(ann2)
+
+
+def test_annotation_eq_false() -> None:
+    """Test that two annotations with different geometries are not equal."""
+    ann_1 = Annotation(Polygon([[0, 0], [1, 1], [2, 0]]))
+    ann_2 = Annotation(Polygon([[0, 0], [1, 1], [2, 0], [2, 2]]))
+    assert ann_1 != ann_2
+
+
+def test_annotation_eq_type_mismatch() -> None:
+    """Test equality with another type."""
+    ann_1 = Annotation(Polygon([[0, 0], [1, 1], [2, 0]]))
+    assert ann_1 != "ann_2"
+
+
+def test_annotation_wkt() -> None:
```

Comment:
This seems like a strange test. I don't have any particular problem with it, but there's a whole array of tests that could be done where we make two things in the same way and check some property matches; just curious why we check this equivalence in particular?

Reply:
Mostly for coverage, update to be a bit more useful